client/ayon_houdini/plugins/publish/collect_render_products.py

```diff
@@ -107,10 +108,11 @@ def replace(match):
                 filename = os.path.join(dirname, filename_base)
                 filename = filename.replace("\\", "/")
 
-            assert "#" in filename, (
-                "Couldn't resolve render product name "
-                "with frame number: %s" % name
-            )
+            if "#" not in filename:
+                raise PublishError(
+                    message="Couldn't resolve render product name"
+                            f" with frame number: {name}"
```

Comment:
I feel using the `message=` argument part here is just redundant clutter. For all errors raised usually the first argument is the message - so it's just as clear without naming it here.

```suggestion
                    "Couldn't resolve render product name"
                    f" with frame number: {name}"
```

Reply:
https://github.com/ynput/ayon-houdini/pull/190/commits/47cd40951abb582d2e2db8bd17c5220653723bae